### PR TITLE
Handle new output from --version in latest Dart SDK

### PR DESCRIPTION
**Pull request by kevmoo**
It may contain a date, which needs to be considered in the parser

Fixes https://github.com/dart-lang/pana/issues/628
Prepare for v0.13.7

### CODE REVIEW

**Comment by kevmoo** [comment]
CC @pq

**Comment by sigurdm** [comment]
I think this is triggered by: https://github.com/dart-lang/sdk/commit/8350f423ee928b095249e82a081209fc349290e3

That means the new info in the version string is the channel

File: lib/src/sdk_env.dart [hunk 1]
```diff
@@ -492,8 +492,9 @@ class DartdocResult {
 }
 
 class DartSdkInfo {
-  static final _sdkRegexp =
-      RegExp('Dart VM version:\\s([^\\s]+)\\s\\(([^\\)]+)\\) on "(\\w+)"');
+  static final _sdkRegexp = RegExp(
+    r'Dart VM version:\s([^\s]+)\s\(([^\)]+)\)(\s\(([^\)]+)\))? on "(\w+)"',
```
Comment (sigurdm) [inline lib/src/sdk_env.dart]: I think you need non capturing groups, or the date will no longer be match[3].
```
r'Dart VM version:\s([^\s]+)\s\(([^\)]+)\)(?:\s\((?:[^\)]+)\))? on "(\w+)"'
```

Comment (isoos) [reply lib/src/sdk_env.dart]: We'll also need to handle both the old and the new output (e.g. try the longer/new regex first, then fallback to the old one), or, if this is a breaking change, we'll need to be specific about it.

Comment (kevmoo) [reply lib/src/sdk_env.dart]: I do handle the optional 2nd item – there's a ? in there that's hard to read.

This works on the bots (without the new change) and on my machine (with the change).

A test would be good, though.

**Comment by kevmoo** [comment]
Should I change this PR or land it? Linter is broken. 😢 

@sigurdm

**Comment by sigurdm** [comment]
@jonasfj could you sanity check my update?

File: test/sdk_env_test.dart [hunk 2]
```diff
@@ -13,5 +13,17 @@ void main() {
         'Dart VM version: 2.0.0-dev.49.0 (Wed Apr 18 20:41:36 2018 +0200) on "macos_x64"';
     final sdkInfo = DartSdkInfo.parse(version);
     expect(sdkInfo.version, Version.parse('2.0.0-dev.49.0'));
+    expect(sdkInfo.dateString, 'Wed Apr 18 20:41:36 2018 +0200');
+    expect(sdkInfo.platform, 'macos_x64');
+  });
+
+  test('parsing SDK version  new style', () {
```
Comment (jonasfj) [inline test/sdk_env_test.dart]: Nit: Extra space on title..

**Comment by kevmoo** [comment]
Thanks, @sigurdm !

**Pull request merged**